src/logging_manager/mod.rs

```diff
@@ -0,0 +1,340 @@
+        let config1 = LoggingConfig {
+            logging_directory: Path::new("./test_logs2/"),
+            max_dir_size: 0
+        };
+        let config2 = LoggingConfig {
+            logging_directory: Path::new("./test_logs3/"),
+            max_dir_size: 1000
+        };
+        let config3 = LoggingConfig {
+            logging_directory: Path::new("./test_logs4/"),
+            max_dir_size: 10000
+        };
+
+        // config 1: max dir size is 0, all files should be deleted
+        create_dir(&config1.logging_directory)?;
+        let mut file = File::create(&config1.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config1)?;
+        remove_dir(&config1.logging_directory)?;
+
+        // config 2: max dir size is 1000 bytes, first file should be deleted (second and third remains)
+        create_dir(&config2.logging_directory)?;
+        let mut file = File::create(&config2.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config2.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config2.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config2)?;
+        assert_eq!(false, config2.logging_directory.join(&file1).exists());
+        assert!(config2.logging_directory.join(&file2).exists());
+        assert!(config2.logging_directory.join(&file3).exists());
+        remove_dir_all(&config2.logging_directory)?;
+
+        // config 3: max dir size is huge, no files should be deleted
+        create_dir(&config3.logging_directory)?;
+        let mut file = File::create(&config3.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config3.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config3.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config3)?;
+        assert!(config3.logging_directory.join(&file1).exists());
+        assert!(config3.logging_directory.join(&file2).exists());
+        assert!(config3.logging_directory.join(&file3).exists());
+        remove_dir_all(&config3.logging_directory)?;
+
+        Ok(())
+    }
```

Comment:
We can add this later but one other good test I just thought of would be logging from different threads concurrently.

Reply:
Good integration test, I will try and add this